rbc/stdlib/creation_functions.py

```diff
@@ -19,6 +20,21 @@
 expose = Expose(globals(), 'creation_functions')
 
 
+def _determine_dtype(dtype, fill_value):
+    if dtype is None:
+        # determine from fill_value
+        if fill_value is None:
+            # return the default floating-point data type
+            return types.double
+        else:
+            return asnumbatype.typeof(fill_value)
+    else:
+        if isinstance(dtype, types.UnicodeType):
+            raise errors.RequireLiteralValue(dtype)
+        else:
```

Comment:
This is for the case one calls a creation function and provide a dtype value (i.e. `array_api.zeros(3, dtype='int32')`). When Numba is compiling the code, it will first type the argument as Unicode and if it fails, with a literal value. Since we want the second case to succeed. Hence, the exception is needed.

Reply:
Thanks for the explanation!